### PR TITLE
Insun vggnet & batch normalization 수정코드입니다

### PR DESCRIPTION
**Pull request by hwanginsun**
1. refactoring 진행(함수화, 코드정리)
2. restore관련 코드 수정(weight를 dict형식으로 받아와서 동일한 값 들어가 있는 내용 확인함)
3. weight update방법을 low에서 high api방법으로 바꿔봤으나 restore할 때의 acc가 나오지 않음
4. plt.plot 기능 추가

### CODE REVIEW

**Comment by anthony0727** [comment]
`conv2d`함수의 아래 부분을 수정해야 됩니다.

layer라는 변수는 먼저 Conv2D 객체를 가르키게 됩니다.
그런데 “BN”이면 가르키던 객체를 버리고, BatchNormalization 객체를 가르키게 됩니다.

그러면 Conv2D객체는 메모리 누수로 이어지게 되고, conv2d 함수가 많이 호출된다면 심각한 문제로 이어질수있습니다.
불필요한 instantiation이 없도록 수정해야 합니다.
```python3
layer = tf.layers.Conv2D(filters = units, # number of kernels 
                             kernel_size = [2,2], 
                             strides = [1,1],
                             padding = 'SAME', 
                             activation = tf.nn.relu, 
                             kernel_initializer = initializer_k, 
                             use_bias = True,
                             bias_initializer = initializer_b, 
                             name = name )(input)
    
if model == "BN" :
    layer = tf.layers.BatchNormalization()(layer, training=is_train)
```

**Comment by anthony0727** [comment]
매개변수명으로 `input`을 사용하시면 안됩니다! 예약어 입니다.

**Comment by hwanginsun** [comment]
넵 감사합니다!
